### ISSUE TITLE
The app immediately crashes on the Simulator.

### ISSUE THREAD

**Issue by HassanTaleb90** [qonversion/qonversion-ios-sdk#365]
The app immediately **crashes** on the Simulator, but works just fine on a real device.

Error log: **[general] NSPredicate: Use of NSFunctionExpression with instance of 'NSBundle' is forbidden.**

Qonversion Version: 3.3.1
Xcode Version: 14.3

**Comment by pixelfoxxx** [comment]
Hello, @HassanTaleb90 

Thank you for bringing this issue to our attention. We have reviewed the problem and can confirm that it is currently in our product backlog. Rest assured; this issue does not affect your production users.

We understand that this issue may cause inconvenience, and we apologize for any inconvenience caused. Please note that we will prioritize this issue and work towards fixing it as soon as we get closer to that task. 

Thank you for your patience and understanding.

**Comment by suriksarkisyan** [comment]
Hi, sorry for the late reply and thank you for bringing this GitHub issue to our attention. After carefully reviewing the details, we have conducted thorough investigations and determined that the problem appears to be a very rare, device-specific case. It is possible that it could be related to a jailbreak version.
To address this issue appropriately, we have created a task and added it to our product backlog and will close this issue for now.

**Comment by suriksarkisyan** [comment]
@HassanTaleb90 could you please give me all details you have? Simulator version and OS, flow to reproduce the issue, etc

**Comment by HassanTaleb90** [comment]
> @HassanTaleb90 could you please give me all details you have? Simulator version and OS, flow to reproduce the issue, etc

Simulator: iPhone 14 Pro Max (iOS 16.4)
Mac OS: Ventura 13.3.1 (a)
Xcode: Version 14.3.1
code:
```
let config = Qonversion.Configuration(projectKey: QonversionManager.projectKey, launchMode: .subscriptionManagement)
		config.setEnvironment(.production)
		config.setEntitlementsCacheLifetime(.month)
		Qonversion.initWithConfig(config)
		Qonversion.shared().setPromoPurchasesDelegate(self)
		Qonversion.shared().collectAdvertisingId()
```
crash:
<img width="1391" alt="Screenshot 2023-06-06 at 2 19 50 PM" src="https://github.com/qonversion/qonversion-ios-sdk/assets/49780321/784b71c4-c53f-46ae-aecd-4a83b5784a9a">

**Comment by suriksarkisyan** [comment]
Thanks a lot! I'll comeback ASAP

**Comment by HassanTaleb90** [comment]
> @HassanTaleb90, could you please also clarify, how you import Qonversion into your project?

1) Using SPM
2) Inside AppDelegate, using
```
func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil
) -> Bool
```

**Comment by SpertsyanKM** [comment]
Strange, but no crash happens even launching your project. I've tried both simulators: iPhone 14 Pro Max (iOS 16.4), and iPhone 14 Pro (iOS 16.4). I build the app on the same OS version via the same XCode and it successfully launches.

**Comment by HassanTaleb90** [comment]
Same issue - Qonversion Version: 4.0.0

**Comment by suriksarkisyan** [comment]
Hi, @HassanTaleb90 
I tried to reproduce the issue, but I couldn't. I added our SDK to the project using SPM. I tested it on the swift project and obj-c project. But I still can't reproduce the problem. Could you give me some details that can be helpful to get and fix the crash?

**Comment by HassanTaleb90** [comment]
The issue was resolved after updating Qonversion to version 5.0.0
